crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+    }
+
+    fn consume_step(&mut self) {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .consume_step();
+    }
+
+    fn get_n_steps(&self) -> Option<usize> {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .get_n_steps()
+    }
+
+    fn run_resources(&self) -> &RunResources {
+        self.blockifier_syscall_handler
+            .context
+            .vm_run_resources
+            .run_resources()
+    }
+}
+
+impl HintProcessorLogic for CairoHintProcessor<'_> {
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        let maybe_extended_hint = hint_data.downcast_ref::<Hint>();
+        if let Some(Hint::Starknet(StarknetHint::Cheatcode {
+            selector,
+            input_start,
+            input_end,
+            output_start,
+            output_end,
+        })) = maybe_extended_hint
+        {
+            return self.execute_cheatcode_hint(
+                vm,
+                exec_scopes,
+                selector,
+                input_start,
+                input_end,
+                output_start,
+                output_end,
+            );
+        }
+        self.blockifier_syscall_handler
+            .execute_hint(vm, exec_scopes, hint_data, constants)
+    }
+
+    /// Trait function to store hint in the hint processor by string.
+    fn compile_hint(
+        &self,
+        hint_code: &str,
+        _ap_tracking_data: &ApTracking,
+        _reference_ids: &HashMap<String, usize>,
+        _references: &[HintReference],
+    ) -> Result<Box<dyn Any>, VirtualMachineError> {
+        Ok(Box::new(self.hints[hint_code].clone()))
+    }
+}
+
+impl CairoHintProcessor<'_> {
+    #[allow(clippy::trivially_copy_pass_by_ref, clippy::too_many_arguments)]
+    pub fn execute_cheatcode_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        _exec_scopes: &mut ExecutionScopes,
+        selector: &BigIntAsHex,
+        input_start: &ResOperand,
+        input_end: &ResOperand,
+        output_start: &CellRef,
+        output_end: &CellRef,
+    ) -> Result<(), HintError> {
+        // Parse the selector.
+        let selector = &selector.value.to_bytes_be().1;
+        let selector = std::str::from_utf8(selector).map_err(|_| {
+            HintError::CustomHint(Box::from(
+                "Failed to parse the  cheatcode selector".to_string(),
+            ))
+        })?;
+
+        // Extract the inputs.
+        let input_start = extract_relocatable(vm, input_start)?;
+        let input_end = extract_relocatable(vm, input_end)?;
+        let inputs = vm_get_range(vm, input_start, input_end).map_err(|_| {
+            HintError::CustomHint(Box::from("Failed to read input data".to_string()))
+        })?;
```

Comment:
This code for error handling is duplicated in couple of places. A helper or some refactor to not do that would be useful.

Reply:
you mean something like this? https://github.com/foundry-rs/starknet-foundry/blob/6722260dbddf56e309e3c549317a296ac9a017cd/crates/cheatnet/src/execution/contract_execution_syscall_handler.rs#L20-L29

Reply:
could be placed in some common crate along with https://github.com/foundry-rs/starknet-foundry/pull/1067#discussion_r1392818816

Reply:
good idea, let's do this under said issue, in separate PR